src/Main.java

```diff
@@ -2257,7 +2374,21 @@ else if (level.equals("W3")) {
                             || (difficulty >= Difficulty.HARD && canGP(inventory) && canLift(inventory) && inventory.contains(Items.JUMP_BOOTS));
                 }
                 else if (location == 1) {
-                    return canGP(inventory);
+					/*
+					* Additional HARD Logic added for this check, which is Main Area - Top Center.
+					* With Flippers or Beanstalk Seeds, Glove, and Boots, you can access this without Overalls.
+					* If no Beanstalk Seeds, Swim beyond the first two sets of pipes in the main area,
+					* then do a midair enemy bounce using the Paragoom.
+					* If you have Beanstalk Seeds, climb the beanstalk, then immediately fall down.
+					* Then go across the 2nd set of pipes and do the same midair enemy bounce using the Paragoom.
+					* The latter option can only be in logic if this check is specifically the Grey or Red Keys.
+					* This is because you would not be able to swim down if it's the Green or Blue Keys in this situation.
+					*/
+                    return canGP(inventory)
+						|| (difficulty >= Difficulty.HARD
+							&& (canSwim(inventory) || (inventory.contains(Items.BEANSTALK_SEEDS) && (keyColor == 0 || keyColor == 1)))					keyColor == 1)
```

Comment:
I did use copy and paste yes, but I don't know what happened there. I removed the extra part. I actually also had to move one of the `)` because it was actually in the wrong spot. Hard logic needs to check for all of these, it was previously only checking the first one.